File: github-copilot/claude-3.7-sonnet/frontend/src/utils/validators.ts
export const validateEmail = (email: string): boolean => {
    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
    return emailRegex.test(email);
};

export const validatePassword = (password: string): boolean => {
    return password.length >= 6;
};

export const validateVocabularyListTitle = (title: string): boolean => {
    return title.trim().length > 0;
};

export const validateFlashCardContent = (content: string): boolean => {
    return content.trim().length > 0;
};

File: github-copilot/claude-3.7-sonnet/backend/api/utils/validators.py
def validate_vocabulary_list_title(title: str) -> bool:
    if not title or len(title) < 3:
        return False
    return True

def validate_flashcard_content(content: str) -> bool:
    if not content or len(content) < 1:
        return False
    return True

def validate_tags(tags: list) -> bool:
    if not isinstance(tags, list):
        return False
    return all(isinstance(tag, str) for tag in tags) and len(tags) > 0